src/compute/src/metrics.rs

```diff
@@ -18,7 +18,7 @@ use prometheus::core::{AtomicF64, GenericCounter};
 pub struct ComputeMetrics {
     // command history
     history_command_count: raw::UIntGaugeVec,
-    history_dataflow_count: UIntGauge,
+    history_dataflow_count: raw::UIntGaugeVec,
```

Comment:
Can you add a comment why it's fine not to use the `DeleteOnDropWrapper` here? The reason is that we never reuse a `clusterd` process, but terminate between new instantiations of `ComputeMetrics`, and the set of labels is fixed at startup time and cannot change.

Alternatively, we should really prevent access to the `raw` metric variants and enforce the `DeleteOnDropWrapper` everywhere. It's so easy to miss the memory leak...

Reply:
Yeah, these are all metrics with timeseries that remain constant throughout the lifetime of the process (except for the per-collection metrics, which use the `DeleteOnDropWrapper`). I'll add a comment.

> Alternatively, we should really prevent access to the `raw` metric variants and enforce the `DeleteOnDropWrapper` everywhere. It's so easy to miss the memory leak...

I remember that I tried using the delete-on-drop variants here and found it quite a hassle for some reason, which is why I switched to the raw variants. I don't remember what the issue was exactly, but I believe it had to do with the fact that you have to create and keep around `DeleteOnDropWrapper` objects for every metric for no good reason.